Sources/System/MachPort.swift

```diff
@@ -0,0 +1,327 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+#if $MoveOnly && (os(macOS) || os(iOS) || os(watchOS) || os(tvOS))
```

Comment:
Does this need to have any interaction with required tools version for the package manifest, or does this just gracefully work out? @glessard do you know?

Reply:
The `#if $SomethingFeature` (magic) notation works as expected on every swift compiler, taking advantage of an unintentional loophole. It is supposed to be superseded by `#if hasFeature(somethingFeature)` in 5.8 and later.

Reply:
Sorry, I was wrong. It only works since 5.3. It should be `#if compiler(>=5.3) && $MoveOnly`.

Reply:
So I went and asked Doug about this. The $MoveOnly syntax works back to the beginning of Swift as long as you are not using new syntax. If we use new syntax, then we need the compiler version there (this is because we still parse the code in the `#if` even if we are eventually going to throw it away). We aren't using new syntax here I think, so I don't think we need the compiler check. But if we eventually do introduce new syntax that can't be parsed by earlier compilers, then we would need the check.

Reply:
The `$` syntax is supposed to make this palatable to at least _some_ older Swift compilers; however, we really should bump the required tools version from 5.2 to at least 5.5 or 5.6 in the next feature release.

If there are any issues, they should show up during manual prerelease testing.

Note: We won't be able to ship a package release that includes `$MoveOnly` features until we have an actual compiler release that supports them. (We cannot consider noncopyable types to be a thing until then.)

Reply:
Hmm, strangely [this broke building with the Swift 5.7.2 compiler, which I just noticed on my Android CI](https://github.com/buttaface/swift-android-sdk/actions/runs/3764473555/jobs/6398952221#step:17:25), though it works with the first 5.8 snapshot that was just released. I am able to reproduce the error when building natively with 5.7.2 for Ubuntu x86_64 too, so this will need to be modified, @loffgren. I notice the CI here only tests with trunk, not with the release 5.7 compiler, so presumably that's how this was missed.